components/calendar/src/any_calendar.rs

```diff
@@ -881,6 +790,101 @@ impl fmt::Display for AnyCalendarKind {
     }
 }
 
+/// TODO
```

Comment:
Observation: New API

Issue: Please add docs since this is landing directly into a stable crate, or open a follow-up issue in the 2.2 milestone.

Reply:
It's not public yet, but will be in https://github.com/unicode-org/icu4x/pull/7314

Reply:
I would like to fully decouple the publicness of this macro from adding the macro.

One very valid route is that we do not make this macro public but we make a handful of AnyCalendar-like types that contain useful collections for users.

Another very valid route is that we do make this macro public but we tweak it to not need this type of trait. Or we rename the trait. I don't want to have to think about that here, I would prefer landing this so we can iterate on top of it.